### PR TITLE
Scale down cluster when max nodes is reduced

### PR DESCRIPTION
**Pull request by manojlds**

### CODE REVIEW

File: src/main/scala/in/ashwanthkumar/vamana2/apps/HadoopScalar.scala [hunk 1]
```diff
@@ -30,12 +30,13 @@ class HadoopScalar extends Scalar[HDemand, HSupply] {
...
     log.info(s"Supply found is $supply")
-    // If the demand is nothing, scale down to cluster min size
-    // If the cluster is running with min capacity and demand > supply, scale it up to max size
-    // else keep the cluster intact
-    if (demand.quantity == 0.0) ctx.cluster.minNodes
-    else if (demand.quantity > supply.available && ctx.currentSize < ctx.cluster.maxNodes) ctx.cluster.maxNodes
-    else ctx.currentSize
+
+    if (demand.quantity == 0.0)
+      return ctx.cluster.minNodes
```
Comment (ashwanthkumar) [inline src/main/scala/in/ashwanthkumar/vamana2/apps/HadoopScalar.scala]: @manojlds I prefer not to use `return` in scala[1]. Can we re-write this without using it? 

[1] - https://tpolecat.github.io/2014/05/09/return.html

**Comment by ashwanthkumar** [comment]
@manojlds except for that one comment, rest LGTM.

File: src/main/scala/in/ashwanthkumar/vamana2/apps/HadoopScalar.scala (hunk 1 above)
Comment (manojlds) [inline src/main/scala/in/ashwanthkumar/vamana2/apps/HadoopScalar.scala]: @ashwanthkumar ok, updated.

**Comment by ashwanthkumar** [comment]
Thanks @manojlds

**Pull request merged**